blockmanager.go

```diff
@@ -221,6 +238,17 @@ func (b *blockManager) resetHeaderState(newestHash *wire.ShaHash, newestHeight i
 	}
 }
 
+// RelayTxDisabled returns whether or not relaying of transactions for the given
+// peer is disabled.
+// It is safe for concurrent access.
+func (b *blockManager) RelayTxDisabled(p *peer.Peer) bool {
+	pd := b.peers[p.Id()]
```

Comment:
Thanks, yeah calling this before initializing the peer in `handleNewPeerMsg` will panic, let me add the check. Also agreed about the mutex map, let me add that.